module/VuFind/src/VuFind/Search/NewItemsHelper.php

```diff
@@ -21,70 +21,65 @@
  * <https://www.gnu.org/licenses/>.
  *
  * @category VuFind
- * @package  Controller_Plugins
+ * @package  Search
  * @author   Demian Katz <demian.katz@villanova.edu>
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org Main Page
  */
 
-namespace VuFind\Controller\Plugin;
+namespace VuFind\Search;
 
-use Laminas\Mvc\Controller\Plugin\AbstractPlugin;
 use Laminas\Mvc\Plugin\FlashMessenger\FlashMessenger;
-use VuFind\Config\Config;
+use VuFind\ILS\Connection;
 
 use function array_slice;
 use function count;
 use function intval;
 use function is_string;
 
 /**
- * Action helper to perform new items-related actions
+ * Helper to perform new items-related actions
  *
  * @category VuFind
- * @package  Controller_Plugins
+ * @package  Search
  * @author   Demian Katz <demian.katz@villanova.edu>
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org Main Page
  */
-class NewItems extends AbstractPlugin
+class NewItemsHelper
 {
-    /**
-     * Configuration
-     *
-     * @var Config
-     */
-    protected $config;
-
     /**
      * Constructor
      *
-     * @param Config $config Configuration
+     * @param array      $config  Configuration
+     * @param Connection $catalog ILS connection
      */
-    public function __construct(Config $config)
+    public function __construct(protected array $config, protected Connection $catalog)
```

Comment:
It also looks like the factory still needs to be modified use getConfigArray instead of getConfigObject and to inject the ILS Connection object.